### PR TITLE
Drop hmac-ripemd160 sshd mac from strong MACs list

### LINKED ISSUE #8212: hmac-ripemd160 no longer available on openssh 7.6 and newer (2017+)
#### Description of problem:

linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs remediation tries to inject a "hmac-ripemd160" hmac.

this is not available on SLES 15.

It was removed with openssh 7.6 upstream in 2017, so it probably can go away here too.
https://www.openssh.com/txt/release-7.6

#### SCAP Security Guide Version:

0.1.60

#### Operating System Version:

SUSE Linux Enterprise Server 15 SP2

#### Steps to Reproduce:

1. run cis bash remediaton
2. restart sshd
3.
4.

#### Actual Results:

sshd no longer starts, complains about "MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,hmac-ripemd160" line

#### Expected Results:

sshd starts

### PR DESCRIPTION
**Pull request by teacup-on-rockingchair**
#### Description:

- Handle different ssh strong macs list for different platforms

#### Rationale:

- Add variable file sshd_strong_macs.var and rework oval to use it when checking ssh configuration
 - Add variable definitions default and cis_sle12, which keeps legacy list and cis sle15, which drops hmac-ripemd160 algorithm, which is no longer supported
 - Add bash and ansible relevant remediations' changes

- Fixes #8212 
#### Review Hints:

- The implementation is borrowed from the approved_macs oval/bash/asible, if considered valuable will extract the command code in a tempmlate, but didn't think it is worthed

### CODE REVIEW

File: linux_os/guide/services/ssh/sshd_strong_macs.var
Comment (jan-cerny) [inline linux_os/guide/services/ssh/sshd_strong_macs.var:16]: missing end of life

File: controls/cis_sle12.yml
Comment (jan-cerny) [inline controls/cis_sle12.yml:1666]: The rule `sshd_use_strong_macs` is used also in multiple other profiles in other products. You should explicitly assign the value of the new `sshd_strong_macs` variable in all the controls and profile files where the rule  `sshd_use_strong_macs` is selected.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (jan-cerny) [inline linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:5]: Is this addition expected? The rule didn't previously depend on FIPS certification and the rule.yml doesn't mention anything about FIPS.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (teacup-on-rockingchair) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:2]: Couldn't find a nice way to transform the comma separated lilst with pipe separated list so can directly use the macro, and didn't felt right to transform the macro that is used in other places.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (teacup-on-rockingchair) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml]: I saw references to the FIPS-140-2 standard in the CIS spec, dropped it in 98293154c6f0efc802f86ff99dfa87324943183a

File: controls/cis_sle12.yml
Comment (teacup-on-rockingchair) [reply controls/cis_sle12.yml:1666]: Did relevant changes in 40cf036f4dcc4877489f4bca239d443a6ad1d3cc but I guess that will need extra review from @ComplianceAsCode/oracle-maintainers,    @ComplianceAsCode/red-hatters and @ComplianceAsCode/ubuntu-maintainers if I made the right assumptions

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (dodys) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:2]: I think with jinja you can just call `replace(",", "|")`

Comment (dodys) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:2]: > I think with jinja you can just call `replace(",", "|")`

nvm, I think it won't work: https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.replace

Comment (dodys) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:2]: maybe with join:
https://jinja.palletsprojects.com/en/2.10.x/templates/#join

File: products/ol7/profiles/e8.profile
Comment (jan-cerny) [inline products/ol7/profiles/e8.profile:133]: The CI fail looks legit, you will have to update the profile stability tests.

Comment (teacup-on-rockingchair) [reply products/ol7/profiles/e8.profile:133]: Should be ok in 7db8bb6

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/bash/shared.sh
Comment (teacup-on-rockingchair) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/bash/shared.sh:3]: Should be covered in 40cf036f4dcc4877489f4bca239d443a6ad1d3cc

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (teacup-on-rockingchair) [reply linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:2]: Yes I did but just getting a bar separated list doesn't get me much further since there is need to do much more changes, which I am afraid cause much more noise in the macros, so decided it doesn't worth at this stage

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (dodys) [inline linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:22]: could you please remove 'rpm' from the comment, different distros use different packaging schemas.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (dodys) [inline linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:25]: could you please remove 'rpm' from the comment, different distros use different packaging schemas.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (dodys) [inline linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:14]: could you please remove 'rpm' from the comment, different distros use different packaging schemas.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml
Comment (dodys) [inline linux_os/guide/services/ssh/ssh_server/sshd_use_strong_macs/oval/shared.xml:11]: could you please remove 'rpm' from the comment, different distros use different packaging schemas.

**Review by dodys — CHANGES_REQUESTED** [review]
minor change needed, but overall looks good

**Comment by dodys** [comment]
@marcusburghardt could you take a look at the packit issues?
the automatus issues is just because this rule is not in their benchmarks

**Comment by marcusburghardt** [comment]
> @marcusburghardt could you take a look at the packit issues? the automatus issues is just because this rule is not in their benchmarks

We are investigating this @dodys . It is not in the project itself but seems to be in the packit side. When it is fixed, we will probably need to re-trigger the failed tests and it should work.

**Comment by teacup-on-rockingchair** [comment]
@dodys and @jan-cerny am I missing something that needs to be done here in order pr to be merged ?

**Review by dodys — APPROVED** [review]
lgtm

**Comment by jan-cerny** [comment]
@ComplianceAsCode/oracle-maintainers @ComplianceAsCode/suse-maintainers PTAL

**Review by jan-cerny — APPROVED** [review]

**Comment by teacup-on-rockingchair** [comment]
> @ComplianceAsCode/oracle-maintainers @ComplianceAsCode/suse-maintainers PTAL

For now I am the only member of  @ComplianceAsCode/suse-maintainers so I have my own approval ;)

**Comment by jan-cerny** [comment]
@freddieRv ping

**Review by freddieRv — APPROVED** [review]

**Comment by teacup-on-rockingchair** [comment]
> @freddieRv ping

Thanks @jan-cerny and @freddieRv  . @jan-cerny can you please overwrite the @ComplianceAsCode/suse-maintainers requirement, since I am the only member for now

**Review by marcusburghardt — APPROVED** [review]
Thanks

**Comment by marcusburghardt** [comment]
Overriding CODEOWNERS as @teacup-on-rockingchair can't approve his own PR.

**Pull request merged**
